test/factory/customSmartWalletFactory.test.ts

```diff
@@ -0,0 +1,537 @@
+        await expect(smartWallet.isInitialized()).to.eventually.be.true;
+      });
+
+      it('should initialize the smart wallet in the expected address paying fee', async function () {
+        const amountToPay = 500;
+
+        const deployRequest = createDeployRequest(
+          {
+            from: owner.address,
+            tokenContract: token.address,
+            tokenAmount: amountToPay,
+            tokenGas: 55000,
+            recoverer: recoverer,
+            index: index,
+            relayHub: worker.address,
+          },
+          {
+            callForwarder: customSmartWalletFactory.address,
+          }
+        );
+
+        const { suffixData, signature } = signDeployRequest(
+          owner,
+          deployRequest,
+          customSmartWalletFactory.address,
+          chainId
+        );
+
+        const initialWorkerBalance = await token.balanceOf(worker.address);
+
+        await customSmartWalletFactory
+          .connect(worker)
+          .relayedUserSmartWalletCreation(
+            deployRequest.request,
+            suffixData,
+            worker.address,
+            signature
+          );
+
+        const smartWallet = await ethers.getContractAt(
+          'SmartWallet',
+          smartWalletAddress
+        );
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(
+          initialWorkerBalance.add(amountToPay)
+        );
+        await expect(smartWallet.isInitialized()).to.eventually.be.true;
+      });
+
+      it('should fail with tokenGas equals to zero while paying fee', async function () {
+        const amountToPay = 500;
+
+        const deployRequest = createDeployRequest(
+          {
+            from: owner.address,
+            tokenContract: token.address,
+            tokenAmount: amountToPay,
+            tokenGas: 0,
+            recoverer: recoverer,
+            index: index,
+            relayHub: worker.address,
+          },
+          {
+            callForwarder: customSmartWalletFactory.address,
+          }
+        );
+
+        const { suffixData, signature } = signDeployRequest(
+          owner,
+          deployRequest,
+          customSmartWalletFactory.address,
+          chainId
+        );
+
+        const initialWorkerBalance = await token.balanceOf(worker.address);
+
+        await expect(
+          customSmartWalletFactory
+            .connect(worker)
+            .relayedUserSmartWalletCreation(
+              deployRequest.request,
+              suffixData,
+              worker.address,
+              signature
+            )
+        ).to.be.rejectedWith('Unable to initialize SW');
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(initialWorkerBalance);
+      });
+
+      it('should fail when owner does not have funds to pay', async function () {
```

Comment:
Why the owner doesn't have funds here but it has in the previous test. I can't find where are you funding it.

Reply:
updated